pandas/compat/__init__.py

```diff
@@ -343,3 +343,14 @@ def is_platform_mac():
 
 def is_platform_32bit():
     return struct.calcsize("P") * 8 < 64
+
+
+class IterableIterator:
+    """Subclass this and provide a "__next__()" method to obtain an iterator.
+    Useful only when the object being iterated is non-reusable (e.g. OK for a
+    parser, not for an in-memory table)."""
+    def __iter__(self):
```

Comment:
you can also incorporate elements of ``TableIterator`` in ``io/pytables.py`` (and make this a sub-class)